src/v/finjector/stress_fiber.cc

```diff
@@ -67,12 +67,33 @@ stress_payload::stress_payload(stress_config cfg) {
     }
 }
 
-[[gnu::noinline]]
-static void spinner(int depth, int spins_per_scheduling_point) { // NOLINT
-    if (depth > 0) {
-        spinner(depth - 1, spins_per_scheduling_point);
+/**
+ * @brief Spins a given number of times at a given stack depth.
+ *
+ * The spinning function will have this many extra frames
+ * added to its stack depth using recursive calls. Each frame is one
+ * of two variants (corresponding to 0 or 1 for the template parameter).
+ * The exact series of frames is set by the seed parameter: each unique seed has
+ * a unique series of calls. This can help stress the CPU profiler by creating a
+ * large number of unique stacks.
+ * @param stack_depth
+ * @param spins_per_scheduling_point number of spins before returning
+ * @param seed the pattern of stack traces will be unique to the seed
+ */
+template<int VALUE>
+[[clang::noinline]] void spinner(
+  int stack_depth, int spins_per_scheduling_point, uint64_t seed) { // NOLINT
+    volatile int spins = 0;
+    // fmt::print("Spinner at depth {}\n", depth);
```

Comment:
remove

Reply:
Removing (artifact of when I was pulling my hair out trying to understand why this wasn't working, answer was: taill call optimization, which of course got disabled by logging here).